src/box/box.cc

```diff
@@ -1686,13 +1686,13 @@ box_promote(void)
 	return rc;
 }
 
-void
+int
 box_listen(void)
 {
 	const char *uri = cfg_gets("listen");
-	if (box_check_uri(uri, "listen") < 0)
-		diag_raise();
-	iproto_listen(uri);
+	if (box_check_uri(uri, "listen") != 0 || iproto_listen(uri) != 0)
```

Comment:
Nit: personally I'd prefer putting each check under separate if:
if (box_check_uri() != 0)
    return -1;
if (iproto_listen(uri) != 0)
    return -1;
    
I believe it's the matter of taste, so you can skip this comment.